### ISSUE TITLE
question: VRAM requirements for SDXL 1024 or another resolution

### ISSUE THREAD

**Issue by blacklig** [HighCWu/control-lora-v3#7]
Hi, your work looks really promissing and I am preparing to try training this. 

But can you tell me why in your example you use 512x512 even for SDXL? Or does it maybe not matter to use this lower res for final output? Or should I aim to use 1024x1024?

Do you know what would be VRAM usage for such?

And can I use different aspect ratios for training as well? Like 1024x768 etc?

Thanks

**Comment by HighCWu** [comment]
Of course, the 512px training will affect the output of sdxl, but my graphics card is only suitable for running 512px. 1024px may require gradient accumulation and gradient_checkpointing techniques to run on a graphics card with 16GB VRAM. For special ratios, such as 1024x768, you need to write dataset-related code yourself.

**Comment by blacklig** [comment]
That is interesting.. I was not able to run full SDXL traininig without offloading to CPU via Deepspeed on my 4090, but with those technics, it is quite slow, would take probably months to train CN.. so your approach with lower res might be good solution.

But how would you compare training SDXL with 512x512 to just training CN on SD 1.5?

I was training SD 1.5 and problem was it was noticable model doesnt have such big context as SDXL, but on SDXL I was able to run only kohyass CN LLLite, which took advantages of bigger context of SDXL but was lacking preision because this LLLite version strips so much weights that in some ways it did not worked at all...

So your Control Lora might be perfect solution but would like to understand it more before I invest time into it, mostly preparing dataset is always biggest hassle :)

**Comment by HighCWu** [comment]
The reason for training sdxl 512x512 is that the generation effect of sdxl is better than sd 1.5. However, I am not sure how much the 512x512 resolution affects the generation effect of sdxl
